src/main/scala/com/linagora/openpaas/gatling/calendar/CalendarSteps.scala

```diff
@@ -0,0 +1,138 @@
+package com.linagora.openpaas.gatling.calendar
+
+import com.linagora.openpaas.gatling.Configuration.{CalendarCount, CalendarSpaPath, SabreBaseUrl}
+import com.linagora.openpaas.gatling.calendar.CalendarConstants.ESNToken
+import com.linagora.openpaas.gatling.utils.RandomUuidGenerator.randomUuidString
+import com.linagora.openpaas.gatling.provisionning.SessionKeys._
+import com.linagora.openpaas.gatling.calendar.SessionKeys._
+import com.linagora.openpaas.gatling.calendar.utils.CalendarUtils
+import com.linagora.openpaas.gatling.core.{DomainSteps, LoginSteps, StaticAssetsSteps, TokenSteps, UserSteps, WebSocketSteps}
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import io.gatling.http.request.builder.HttpRequestBuilder
+import io.gatling.core.structure.ChainBuilder
+
+import scala.concurrent.duration.DurationInt
+import scala.util.matching.Regex
+
+object CalendarSteps {
+  def openCalendarSPA(): ChainBuilder = {
+    group("openCalendarSPA") {
+      exec(LoginSteps.loadLoginTemplates)
+        .exec(LoginSteps.login())
+        .exec(StaticAssetsSteps.loadIndexHtmlAndMainJs(CalendarSpaPath))
+        .exec(StaticAssetsSteps.loadStaticAssets(CalendarStaticAssets.OpeningCalendarAssets))
+        .exec(TokenSteps.retrieveAuthenticationToken)
+        .exec(WebSocketSteps.getSocketId)
+        .exec(WebSocketSteps.registerSocketNamespaces)
+        .exec(WebSocketSteps.openWsConnection())
+        .exec(UserSteps.getProfile())
+        .exec(DomainSteps.getDomain)
+        .exec(DomainSteps.getThemeForDomain())
+        .exec(DomainSteps.getLogoForDomain)
+        .exec(CalendarSteps.getCalendarConfiguration())
+        .exec(CalendarSteps.listUsableCalendarsForUser())
+        .exec(EventSteps.listEvents())
+    }
+  }
+
+  def createCalendar(): HttpRequestBuilder = {
+    val calId: String = randomUuidString
+
+    http("createCalendar")
+      .post(s"$SabreBaseUrl/calendars/$${$UserId}")
+      .header(ESNToken, s"$${$Token}")
+      .body(StringBody(s"""
+        {
+        "id":"$calId",
```

Comment:
the callId will always be the same.
you should do 
```
body(StringBody(session => {
//generate new uid here
}
```

Reply:
Is this what you mean?

```

Reply:
The UUID generator has something to do with session?

Reply:
No but if you don't do it in a method taking the current session as parameter, it will be generated when constructing the scenario and will be hard coded in the body of your request.